packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,493 @@
+      y: pivotYCorrd,
+      node: pivotDataReference,
+      onTap: () => cellOnTap(pivotDataReference.name),
+      levelsVisible: widget.levelsVisible - 1,
+    );
+
+    // Construct list 3 sub-treemap.
+    final double list3Ratio = list3Size / totalSize;
+    final double list3Width =
+        isHorizontalRectangle ? list3Ratio * width : width;
+    final double list3Height =
+        isHorizontalRectangle ? height : list3Ratio * height;
+    final double list3XCoord =
+        isHorizontalRectangle ? x + list1Width + pivotBestWidth : x;
+    final double list3YCoord =
+        isHorizontalRectangle ? y : y + list1Height + pivotBestHeight;
+    List<Cell> list3Cells;
+    list3Cells = buildTreemap(
+      list3,
+      list3Width,
+      list3Height,
+      list3XCoord,
+      list3YCoord,
+    );
+
+    return list1Cells + [pivotCell] + list2Cells + list3Cells;
+  }
+
+  String nodeSizeText() {
+    final int size = rootNode.byteSize;
+    final double sizeInKB = size / 1024;
+    if (sizeInKB < 1024.0) {
+      return '[' + sizeInKB.toStringAsFixed(2) + ' KB]';
+    } else {
+      // TODO(peterdjlee): Investigate the max size of memory usages.
+      final double sizeInMB = sizeInKB / 1024;
+      return '[' + sizeInMB.toStringAsFixed(2) + ' MB]';
+    }
+  }
+
+  // Treemap layout:
+  // ----------------------------
+  // |        Title Text        |
+  // |--------------------------|
+  // |                          |
+  // |         Content          |
+  // |                          |
+  // |                          |
+  // ----------------------------
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: EdgeInsets.all(widget.levelsVisible > 0 ? 1.0 : 0.0),
+      child: Container(
+        decoration: BoxDecoration(border: Border.all(width: 0.5)),
+        child: Column(
+          children: [
+            if (widget.levelsVisible > 0 &&
+                rootNode.children.isNotEmpty &&
+                shouldDisplayText)
+              Center(
+                child: Text(
+                  rootNode.name + ' ' + nodeSizeText(),
+                  overflow: TextOverflow.ellipsis,
+                  textAlign: TextAlign.center,
+                ),
+              ),
+            Expanded(child: buildContent()),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildContent() {
+    if (widget.levelsVisible == 0) {
+      if (shouldDisplayText) {
+        return Container(
+          width: widget.width,
+          child: Center(
+            child: nameAndSizeText(),
+          ),
+        );
+      } else {
+        return Container();
+      }
+    } else if (widget.levelsVisible == 1) {
+      if (rootNode.children.isNotEmpty) {
+        return GestureDetector(
+          onTap: widget.onTap,
+          child: Container(
+            color: Colors.white38,
+            child: Tooltip(
+              message: rootNode.name,
+              child: buildNestedTreemap(),
```

Comment:
This should create a TreeMap widget instead of using the methods `buildTreemap` and `buildNestedTreemap`